controllers/dspipeline_controller.go

```diff
@@ -335,11 +337,36 @@ func (r *DSPAReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 	return ctrl.Result{}, nil
 }
 
+func (r *DSPAReconciler) setStatusAsNotReady(conditionType string, err error, setStatus func(metav1.Condition)) {
+	condition := dspastatus.BuildFalseCondition(conditionType, config.FailingToDeploy, err.Error())
+	setStatus(condition)
+}
+
+func (r *DSPAReconciler) setStatusAsReady(ctx context.Context, resourceName string, conditionType string,
+	dspa *dspav1alpha1.DataSciencePipelinesApplication, setStatus func(metav1.Condition),
+	log logr.Logger) {
+	condition, err := r.handleReadyCondition(ctx, dspa, resourceName, conditionType)
+	setStatus(condition)
+	if err != nil {
+		log.Error(err, fmt.Sprintf("Encountered error when creating the %s readiness condition", conditionType))
+	}
+}
```

Comment:
The name of this `setStatusAsReady` is misleading, probably because `handleReadyCondition` could have been named better. If you look inside this function, you will see that the conditions that are returned can actually be conditions that do not mean the component is ready.